src/Sarif.Viewer.VisualStudio/Tags/SarifLocationErrorTagger.cs

```diff
@@ -0,0 +1,139 @@
+    /// that are children of the currently selected <see cref="SarifErrorListItem"/> as defined by <see cref="ISarifErrorListEventSelectionService.SelectedItem"/>.
+    /// This allows us to provide error tags for code flows for a specific result without displaying potentially overlapping code flow error tags from other results.
+    /// </remarks>
+    internal class SarifLocationErrorTagger : ITagger<IErrorTag>, ISarifLocationTagger, IDisposable
+    {
+        private bool isDisposed;
+
+        /// <summary>
+        /// The file path associated with the <see cref="ITextBuffer"/> given in the constructor.
+        /// </summary>
+        private readonly string filePath;
+
+        private readonly IPersistentSpanFactory persistentSpanFactory;
+        private readonly ISarifErrorListEventSelectionService sarifErrorListEventSelectionService;
+        private readonly ITextBuffer textBuffer;
+
+        private List<ISarifLocationTag> currentTags;
+        private bool tagsDirty = true;
+
+        /// <inheritdoc/>
+        public event EventHandler<SnapshotSpanEventArgs> TagsChanged;
+
+        /// <inheritdoc/>
+        public event EventHandler Disposed;
+
+        public SarifLocationErrorTagger(ITextBuffer textBuffer, IPersistentSpanFactory persistentSpanFactory, ISarifErrorListEventSelectionService sarifErrorListEventSelectionService)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (!SdkUIUtilities.TryGetFileNameFromTextBuffer(textBuffer, out this.filePath))
+            {
+                throw new ArgumentException("Always expect to be able to get file name from text buffer.", nameof(textBuffer));
+            }
+
+            this.textBuffer = textBuffer;
+            this.persistentSpanFactory = persistentSpanFactory;
+            this.sarifErrorListEventSelectionService = sarifErrorListEventSelectionService;
+            this.sarifErrorListEventSelectionService.SelectedItemChanged += SarifErrorListEventSelectionService_SelectedItemChanged;
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<ITagSpan<IErrorTag>> GetTags(NormalizedSnapshotSpanCollection spans)
+        {
+            if (this.tagsDirty)
+            {
+                this.tagsDirty = false;
+
+                // This query is saying, create the error tags for the top-level SARIF result (includeChildTags: false, includeResultTag: true)
+                // and then create the tags for the children (includeChildTags: true, includeResultTag: false) but then
+                // filter the children (thread-flows, call nodes, etc.) to the ones relevant to the currently selected error item.
+                // This is done so that duplicate locations from multiple results in results are rendered in the editor.
+                IEnumerable<ISarifLocationTag> possibleTags = CodeAnalysisResultManager.
```

Comment:
>possibleTags [](start = 47, length = 12)

This is declared as IEnumerable, but actually assigned from a list. and then the next statement calls ToList again. Why not just declare `possibleTags` as a List, and assign the result of the query directly to `this.currentTags` (but see my next comment about the formulation of the query).